File: app/lib/domain/entities/statistics.dart
enum ConcentrationPeriod {
  morning('MORNING'),
  afternoon('AFTERNOON');

  final String value;
  const ConcentrationPeriod(this.value);

  static ConcentrationPeriod fromString(String value) {
    switch (value) {
      case 'MORNING':
        return ConcentrationPeriod.morning;
      case 'AFTERNOON':
        return ConcentrationPeriod.afternoon;
      default:
        throw ArgumentError('Invalid ConcentrationPeriod: $value');
    }
  }
}

class CategoryDistribution {
  final String categoryId;
  final String categoryName;
  final int totalFocusTime;
  final double percentage;
  final List<TaskDistribution> taskDistribution;

  CategoryDistribution({
    required this.categoryId,
    required this.categoryName,
    required this.totalFocusTime,
    required this.percentage,
    required this.taskDistribution,
  });
}

class TaskDistribution {
  final String taskName;
  final int totalFocusTime;
  final double percentage;

  TaskDistribution({
    required this.taskName,
    required this.totalFocusTime,
    required this.percentage,
  });
}

class DailyActivityDistribution {
  final String categoryId;
  final String categoryName;
  final int totalFocusTime;

  DailyActivityDistribution({
    required this.categoryId,
    required this.categoryName,
    required this.totalFocusTime,
  });
}

class DailyActivity {
  final int date;
  final List<DailyActivityDistribution> categoryDistribution;

  DailyActivity({required this.date, required this.categoryDistribution});
}

class PeriodStatistics {
  final int totalSessions;
  final int totalBreaks;
  final int totalFocusTime;
  final int totalBreakTime;
  final ConcentrationPeriod mostConcentratedPeriod;
  final ConcentrationPeriod lessConcentratedPeriod;
  final List<int> concentrationDistribution;
  final List<CategoryDistribution> categoryDistribution;
  final List<DailyActivity> dailyActivity;

  PeriodStatistics({
    required this.totalSessions,
    required this.totalBreaks,
    required this.totalFocusTime,
    required this.totalBreakTime,
    required this.mostConcentratedPeriod,
    required this.lessConcentratedPeriod,
    required this.concentrationDistribution,
    required this.categoryDistribution,
    required this.dailyActivity,
  });
}

File: backend/crates/domain/src/entities/stats/statistics.rs
use crate::entities::stats::calculators::concentration_calculator::ConcentrationStats;
use crate::entities::stats::calculators::daily_activity_calculator::DailyActivityItem;
use crate::entities::stats::calculators::period_summary_calculator::PeriodSummary;
use crate::entities::stats::category_distribution::CategoryDistributionItem;
use crate::entities::stats::concentration_period::ConcentrationPeriod;

#[derive(Debug, Clone)]
pub struct Stats {
    total_sessions: usize,
    total_breaks: usize,
    total_focus_time: i64,
    total_break_time: i64,
    most_concentrated_period: ConcentrationPeriod,
    less_concentrated_period: ConcentrationPeriod,
    concentration_distribution: [u32; 5],
    category_distribution: Vec<CategoryDistributionItem>,
    daily_activity: Vec<DailyActivityItem>,
}

impl Stats {
    pub fn new(
        period_summary: PeriodSummary,
        concentration_stats: ConcentrationStats,
        category_distribution: Vec<CategoryDistributionItem>,
        daily_activity: Vec<DailyActivityItem>,
    ) -> Self {
        Self {
            total_sessions: period_summary.total_sessions,
            total_breaks: period_summary.total_breaks,
            total_focus_time: period_summary.total_focus_time,
            total_break_time: period_summary.total_break_time,
            most_concentrated_period: concentration_stats.most_concentrated_period,
            less_concentrated_period: concentration_stats.less_concentrated_period,
            concentration_distribution: concentration_stats.concentration_distribution,
            category_distribution,
            daily_activity,
        }
    }

    pub fn total_sessions(&self) -> usize {
        self.total_sessions
    }

    pub fn total_breaks(&self) -> usize {
        self.total_breaks
    }

    pub fn total_focus_time(&self) -> i64 {
        self.total_focus_time
    }

    pub fn total_break_time(&self) -> i64 {
        self.total_break_time
    }

    pub fn most_concentrated_period(&self) -> &ConcentrationPeriod {
        &self.most_concentrated_period
    }

    pub fn less_concentrated_period(&self) -> &ConcentrationPeriod {
        &self.less_concentrated_period
    }

    pub fn concentration_distribution(&self) -> &[u32; 5] {
        &self.concentration_distribution
    }

    pub fn category_distribution(&self) -> &[CategoryDistributionItem] {
        &self.category_distribution
    }

    pub fn daily_activity(&self) -> &[DailyActivityItem] {
        &self.daily_activity
    }
}
